cryptol-remote-api/Dockerfile

```diff
@@ -43,18 +44,40 @@ RUN latest="$(curl -sSL 'http://cvc4.cs.stanford.edu/downloads/builds/x86_64-lin
 # Set executable and run tests
 RUN chmod +x rootfs/usr/local/bin/*
 
-ARG GHCVER
 FROM debian:buster AS build
-ARG GHCVER
-
+ARG PORTABILITY=false
 RUN apt-get update && apt-get install -y libncurses-dev libz-dev \
-    build-essential curl libffi-dev libffi6 libgmp-dev libgmp10 libncurses-dev libncurses5 libtinfo5 libnuma-dev
+    build-essential curl libffi-dev libffi6 libgmp-dev libgmp10 libncurses-dev libncurses5 libtinfo5 libnuma-dev \
+    $(if ${PORTABILITY}; then echo git autoconf python3; fi)
 RUN curl -o /usr/local/bin/ghcup "https://downloads.haskell.org/~ghcup/0.1.12/x86_64-linux-ghcup-0.1.12" && chmod +x /usr/local/bin/ghcup
 ENV GHCUP_USE_XDG_DIRS=true \
     XDG_BIN_HOME=/usr/local/bin \
     XDG_DATA_HOME=/usr/local/share
 COPY --from=solvers /solvers/rootfs /
-RUN ghcup install ghc ${GHCVER} && ghcup set ghc ${GHCVER} && ghcup install cabal --set
+RUN ghcup install cabal --set
+ENV PATH /root/.cabal/bin:$PATH
+ADD ./cryptol-remote-api/ghc-portability.patch .
+ARG GHCVER
+ARG GHCVER_BOOTSTRAP
+RUN if ${PORTABILITY}; then \
```

Comment:
I think a lot of this ghc install logic can be largely replaced with `ghcup compile ghc -b ${GHCVER_BOOTSTRAP} -v ${GHCVER} -p /patches` (assuming the `ghc-portability.patch` is relocated to `/patches`).

But that's something to explore later rather than delay this.

Reply:
I don't think this is holding anything up, there's already a working version in the registry that we've sent them. Refactoring that now will be less expensive than doing it later. @atomb ?

Reply:
Sure, if you're available and interested to simplify it now, go for it!

Reply:
Looked at it and `ghcup compile` won't work here, since it compiles from source tarballs. The portability patch affects the generated `./configure` scripts via `configure.ac` and so must be applied before autoconf is run (`./boot`). Source tarballs already have the `./configure` files built so applying the patch there wouldn't make a difference.

Reply:
Awesome, good to know. There's a way to do it with ghcup; I think it might be possible to simply reconfigure it, but a custom bindist could certainly be used with the patch already applied; at that point, though, there's not a lot of ergonomics saved.